receiver/kubeletstatsreceiver/internal/kubelet/metadata.go

```diff
@@ -151,3 +223,67 @@ func (m *Metadata) getPodVolume(podUID string, volumeName string) (v1.Volume, er
 
 	return v1.Volume{}, fmt.Errorf("pod %q with volume %q not found in the fetched metadata", podUID, volumeName)
 }
+
+func (m *Metadata) getPodCPULimit(uid string) float64 {
+	podResource, ok := m.PodResources[uid]
+	if !ok {
+		return 0
+	}
+	return podResource.cpuLimit
+}
```

Comment:
No sure if this function is needed now. It's essentially the same as `m.PodResources[uid]`, right?

```suggestion
func (m *Metadata) getPodCPULimit(uid string) float64 {
	return m.PodResources[uid]
}
```

Reply:
We leak a little bit of the key logic for the containers map, but I am ok simplifying for now.